app/src/androidTest/java/ch/epfl/sdp/peakar/points/POICacheTest.java

```diff
@@ -50,25 +49,43 @@
     @Rule
     public GrantPermissionRule grantCameraPermissionRule2 = GrantPermissionRule.grant(Manifest.permission.READ_EXTERNAL_STORAGE);
 
+
+    private static Pair<int[][], Double> topographyPair;
+
     @BeforeClass
-    public static void setup(){
+    public static void setup() throws InterruptedException {
         //Remove potentially created file (due to test suite)
-        deleteCacheFile();
+        POICache.getInstance().deleteCacheFile(ApplicationProvider.getApplicationContext().getCacheDir());
+        Context mContext = ApplicationProvider.getApplicationContext();
+        UserPoint userPoint = UserPoint.getInstance(mContext);
+        userPoint.setLocation(GPSTracker.DEFAULT_LAT, GPSTracker.DEFAULT_LON,GPSTracker.DEFAULT_ALT, GPSTracker.DEFAULT_ACC);
+        new DownloadTopographyTask(){
+            @Override
+            public void onResponseReceived(Pair<int[][], Double> topography) {
+                super.onResponseReceived(topography);
+                topographyPair = topography;
+            }
+        }.execute(userPoint);
+
+        //Wait for the map to be downloaded
+        int counter=0;
+        while(topographyPair==null && counter<20){
+            Thread.sleep(1000);
+            counter++;
+        }
+        counter=0;
+        Assert.assertNotNull(topographyPair);
+        while(topographyPair.first==null && counter<20){
+            Thread.sleep(1000);
+            counter++;
```

Comment:
I don't think we need to wait with a cicle, you could put the assert statement at the end of the onResponseReceived so as soon as it is done it will call it

Reply:
resolved in next commit